ipld/car/src/util.rs

```diff
@@ -44,7 +47,9 @@ pub(crate) async fn ld_write<'a, W>(writer: &mut W, bytes: &[u8]) -> Result<(),
 where
     W: AsyncWrite + Send + Unpin,
 {
-    writer.write_varint_async(bytes.len()).await?;
+    let mut len = unsigned_varint::encode::usize_buffer();
+    let len = unsigned_varint::encode::usize(bytes.len(), &mut len);
+    writer.write_all(len).await?;
```

Comment:
Nit, but would find this easier to read than calling it `len` each time

```suggestion
    let mut buff = unsigned_varint::encode::usize_buffer();
    let len = unsigned_varint::encode::usize(bytes.len(), &mut buff);
    writer.write_all(len).await?;
```